Feat: Support required param type object, boolean

kilianmh opened the pull request:
Now the types "object" and "boolean" are supported for required parameters.
`Boolean` is implemented by creating one method with specializer `null` and
one with `eql-specializer` with object `t`.

Also: 
- change `generate-method-descriptions` function and `docstring` to
allow printing of `eql-specializer`.
- add regression test for `describe-object` and add more methods to `spec.json` for regression tests to cover required parameters of type `array` without items, `boolean` and `object`.

kilianmh committed:
Feat: Support required param type object, boolean

Now the types "object" and "boolean" are supported for required parameters.
Boolean is implemented by creating one method with specializer null and
one with eql-specializer with object t.

Also: change generate-method-descriptions function and docstring to
allow printing of eql-specializer.

kilianmh committed:
Tests: Add regression test for describe-object

Also add more methods to spec.json for regression tests to cover required parameters of
type array without items, boolean and object.

svetlyak40wt commented:
Cool! Clever approach with boolean handling!

This improvement deserve its place in the changelog!

svetlyak40wt commented:
You will need to merge master into your branch before changing changelog.lisp, because I've merged this PR today: https://github.com/40ants/openrpc/pull/17

kilianmh committed:
Docs: Changelog version 0.10.1

- Support object and boolean type for required parameters.
- Add regression tests for generated DESCRIBE-OBJECT method.

kilianmh commented:
Again sorry for forgetting the changelog. Now its rebased on master and added the changes in the docs.

svetlyak40wt commented:
Great!
